lib/logitech_receiver/settings.py

```diff
@@ -356,3 +357,36 @@ def prepare_write(self, new_value, current_value=None):
 			raise ValueError("invalid choice %r" % new_value)
 		assert isinstance(choice, _NamedInt)
 		return choice.bytes(self._bytes_count)
+
+class RangeValidator(object):
+	__slots__ = ('min_value', 'max_value', 'flag', '_bytes_count', 'needs_current_value')
+
+	kind = KIND.range
+
+	"""Translates between integers and a byte sequence.
+	:param min_value: minimum accepted value (inclusive)
+	:param max_value: maximum accepted value (inclusive)
+	:param bytes_count: the size of the derived byte sequence. If None, it
+	will be calculated from the range."""
+	def __init__(self, min_value, max_value, bytes_count=None):
+		assert max_value > min_value
+		self.min_value = min_value
+		self.max_value = max_value
+		self.needs_current_value = False
+
+		self._bytes_count = math.ceil(math.log(max_value + 1, 256))
```

Comment:
Apparently Python has a `.bit_length()` function, see also `ChoicesValidator.__init__` (maybe we should add a helper function to `common.py`?)